[CI] DataStreamDeprecationCheckerTests testOldIndicesCheckWithOnlyNewIndices failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #9040 / ubuntu-2004_platform-support-unix](https://gradle-enterprise.elastic.co/s/dk5pyrr5xyjbw)
- [elasticsearch-periodic-platform-support #9033 / ubuntu-2404-aarch64_checkpart2_platform-support-arm](https://gradle-enterprise.elastic.co/s/ysaqhni4ybrts)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:deprecation:test" --tests "org.elasticsearch.xpack.deprecation.DataStreamDeprecationCheckerTests.testOldIndicesCheckWithOnlyNewIndices" -Dtests.seed=17BD723E3E20BC5 -Dtests.locale=guz -Dtests.timezone=Antarctica/Palmer -Druntime.java=24
```

**Applicable branches:**
8.18

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.deprecation.DataStreamDeprecationCheckerTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testOldIndicesCheckWithOnlyNewIndices),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [8.18] 2 failures in test testOldIndicesCheckWithOnlyNewIndices (1.8% fail rate in 109 executions)
- [8.18] 2 failures in pipeline elasticsearch-periodic-platform-support (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/DataStreamDeprecationCheckerTests.java
+++ b/x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/DataStreamDeprecationCheckerTests.java
@@ -75,7 +75,7 @@ public class DataStreamDeprecationCheckerTests extends ESTestCase {
 
     public void testOldIndicesCheckWithOnlyNewIndices() {
         // This tests what happens when any old indices that we have are closed. We expect no deprecation warning.
-        int newOpenIndexCount = randomIntBetween(0, 100);
+        int newOpenIndexCount = randomIntBetween(1, 100);
         int newClosedIndexCount = randomIntBetween(0, 100);
 
         Map<String, IndexMetadata> nameToIndexMetadata = new HashMap<>();

